packages/social-previews/src/mastodon-preview/helpers.ts

```diff
@@ -0,0 +1,21 @@
+import { firstValid, hardTruncation, shortEnough, stripHtmlTags, Formatter } from '../helpers';
+
+const TITLE_LENGTH = 200;
+const BODY_LENGTH = 500;
+const URL_LENGTH = 30;
+
+export const mastodonTitle: Formatter = ( text ) =>
+	firstValid(
+		shortEnough( TITLE_LENGTH ),
+		hardTruncation( TITLE_LENGTH )
+	)( stripHtmlTags( text ) ) || '';
+
+export const mastodonBody: Formatter = ( text, offset = 0 ) =>
+	firstValid(
+		shortEnough( BODY_LENGTH - URL_LENGTH - offset ),
+		hardTruncation( BODY_LENGTH - URL_LENGTH - offset )
+	)( stripHtmlTags( text ) ) || '';
```

Comment:
Does Mastodon support hashtags or hyperlinks? You may wish to use `preparePreviewText` which I updated in #77432

Reply:
Since that PR is still open, let's handle this separately.

Reply:
Let me know if you'll handle it in your PR, or if we should take care of it.

Reply:
It's merged now. I think it can be done in a separate PR.